### PR TITLE
SYNCT-45: Create ClientFactory

### PR DESCRIPTION
**Pull request by kcissewski**
- Added ClientFactory class
- Changed unit tests to fit new implementation

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/sync2/api/sync/SyncClient.java [hunk 1]
```diff
@@ -20,10 +21,12 @@ public Object pullDataFromParent(String category, Map<String, String> resourceLi
         String preferredClient = Context.getAdministrationService().getGlobalProperty(RESOURCE_PREFERRED_CLIENT);
         String url = address + resourceLinks.get(preferredClient);
 
+        ClientFactory clientFactory = new ClientFactory();
         if (REST_CLIENT_KEY.equals(preferredClient)) {
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/sync2/api/sync/SyncClient.java]: It is a redundant condition.  You use the same if statement in the ClientFactory. IMHO you can use the createClient method only once with parameter "preferredClient"

File: api/src/main/java/org/openmrs/module/sync2/client/rest/RestClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.openmrs.module.sync2.client.rest;
+
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+
+public class RestClientFactory extends HttpComponentsClientHttpRequestFactory {
+
+    private final static int connectionRequestTimeout = 1000;
+    private final static int connectTimeout = 15 * 1000;
+    private final static int readTimeout = 30 * 1000;
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/sync2/client/rest/RestClientFactory.java]: These 3 constants names should be Upper-case with the underscore between words. Additionally, you can use 1000 as const.

File: api/src/main/java/org/openmrs/module/sync2/api/sync/SyncClient.java (hunk 1 above)
Comment (ubap) [reply api/src/main/java/org/openmrs/module/sync2/api/sync/SyncClient.java]: We need some kind of client interface. If Karol didn't use this condition he wouldn't be able to know how to cast the object returned by createClient(...) method.

**Comment by pkornowski** [comment]
The approach will be adjusted further in FM-212.

**Pull request merged**